src/dotnet-core-uninstall/Shared/VSVersioning/VisualStudioSafeVersionsExtractor.cs

```diff
@@ -8,76 +7,88 @@
 
 namespace Microsoft.DotNet.Tools.Uninstall.Shared.VSVersioning
 {
-    // Visual Studio versions 15.3 thru 15.9 (inclusive) requires an sdk with max version 2.1.5xx (inclusive)
-    //                        16.0 requires anything less than 3.0.1xx
-    // Must keep one of each Major.Minor band to ensure runtime works
     internal static class VisualStudioSafeVersionsExtractor
     {
-        // Divisions within version bands. Not inclusive, groupings: [2.1.000, 2.1.600) [2.1.600, 2.2.00) [2.0.00, 2.2.200) [2.2.200, 2.3.00)
-        private static readonly SemanticVersion[] SpecialCaseDivisions = { new SemanticVersion(2, 1, 600), new SemanticVersion(2, 2, 200) };
-
         // The tool should not be used to uninstall any more recent versions of the sdk
-        public static readonly SemanticVersion UpperLimit = RuntimeInfo.RunningOnWindows ? new SemanticVersion(3, 0, 0) : new SemanticVersion(5, 0, 0);
+        public static readonly SemanticVersion UpperLimit = new SemanticVersion(5, 0, 0);
```

Comment:
we change it from 3.0 to 5.0 why?will it try to uninstall the dummy entry Visual Studio created?

Reply:
The upper limit was mostly meant to cap the use of the tool so people don't continue to use it to uninstall future versions. There's still logic to protect the highest version below that upper limit (usually 3.1.xxx or 3.0.xxx) [here](https://github.com/sfoslund/cli-lab/blob/RestructureCommands/src/dotnet-core-uninstall/Shared/VSVersioning/VisualStudioSafeVersionsExtractor.cs#L25). I'm not sure what the Visual Studio dummy entry is, can you give me some more information?